File: app/src/main/java/com/mytlogos/enterprise/background/room/model/RoomUser.java
package com.mytlogos.enterprise.background.room.model;

import androidx.annotation.NonNull;
import androidx.room.Entity;
import androidx.room.PrimaryKey;


@Entity
public class RoomUser {
    @NonNull
    private final String name;

    @NonNull
    @PrimaryKey
    private final String uuid;

    @NonNull
    private final String session;

    public RoomUser(@NonNull String name, @NonNull String uuid, @NonNull String session) {
        this.name = name;
        this.uuid = uuid;
        this.session = session;
    }


    @NonNull
    public String getUuid() {
        return uuid;
    }


    @NonNull
    public String getName() {
        return name;
    }


    @NonNull
    public String getSession() {
        return session;
    }


    public String toString() {
        return super.toString() +
                "{" +
                "uuid: " + this.getUuid() +
                ", name: " + this.getName() +
                "}";
    }

    public int hashCode() {
        return this.getUuid().hashCode();
    }


    public boolean equals(Object obj) {
        if (obj == this) return true;
        if (obj == null) return false;
        if (!(obj instanceof RoomUser)) return false;

        RoomUser user = (RoomUser) obj;
        return this.getUuid().equals(user.getUuid());
    }
}

File: app/src/main/java/com/mytlogos/enterprise/background/room/model/RoomUser.kt
package com.mytlogos.enterprise.background.room.model

import androidx.room.Entity
import androidx.room.PrimaryKey

@Entity
class RoomUser(val name: String, @field:PrimaryKey val uuid: String, val session: String) {

    override fun toString(): String {
        return super.toString() +
                "{" +
                "uuid: " + uuid +
                ", name: " + name +
                "}"
    }

    override fun hashCode(): Int {
        return uuid.hashCode()
    }

    override fun equals(other: Any?): Boolean {
        if (other === this) return true
        if (other == null) return false
        if (other !is RoomUser) return false
        return uuid == other.uuid
    }
}